src/mito2/src/sst/location.rs

```diff
@@ -29,8 +30,48 @@ pub fn index_file_path(region_dir: &str, sst_file_id: FileId) -> String {
     util::join_path(&dir, &sst_file_id.as_puffin())
 }
 
+/// `IntermediateLocation` produces paths for intermediate files
+/// during external sorting.
+#[derive(Debug, Clone)]
+pub struct IntermediateLocation {
+    root_path: String,
+}
+
+impl IntermediateLocation {
+    /// Create a new `IntermediateLocation`. Set the root directory to
```

Comment:
Question: if the user uses S3 for storage, does the intermediate results will be written into S3? Or it will always use the local file disk? I think the later is better for such intermediate results.

Reply:
Okay, I can use parameters to distinguish different Object Stores. One of them is a local file system dedicated to intermediate results.